src/then.js

```diff
@@ -0,0 +1,36 @@
+var _curry2 = require('./internal/_curry2');
+var _isFunction = require('./internal/_isFunction');
+var toString = require('./toString');
+
+
+/**
+ * Returns the result of calling `then` on the supplied promise with the
+ * supplied function. This is useful for working with promises inside
+ * function compositions.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (a -> b) -> Promise a -> Promise b
```

Comment:
I think we need a second signature:

```haskell
then :: (a -> Promise b) -> Promise a -> Promise b
```

Essentially, this function is both `map` and `chain`. :cry: